nova_vm/src/ecmascript/syntax_directed_operations/class_definitions.rs

```diff
@@ -52,11 +85,539 @@ pub(crate) fn derived_class_default_constructor(
     let func = f.internal_get_prototype_of(agent).unwrap();
     // 3. If IsConstructor(func) is false, throw a TypeError exception.
     let Some(func) = func.and_then(|func| is_constructor(agent, func)) else {
-        return Err(agent.throw_exception_with_static_message(ExceptionType::TypeError, "Expected callable function"));
+        return Err(agent.throw_exception_with_static_message(
+            ExceptionType::TypeError,
+            "Expected callable function",
+        ));
     };
     // 4. Let result be ? Construct(func, args, NewTarget).
-    let result = construct(agent, func, Some(args), Some(Function::try_from(new_target).unwrap()))?;
+    let result = construct(
+        agent,
+        func,
+        Some(args),
+        Some(Function::try_from(new_target).unwrap()),
+    )?;
     // vi. Perform ? InitializeInstanceElements(result, F).
     // vii. Return result.
     Ok(result.into_value())
-}
\ No newline at end of file
+}
+
+impl CompileEvaluation for ast::Class<'_> {
+    /// ClassTail : ClassHeritage_opt { ClassBody_opt }
+    fn compile(&self, ctx: &mut CompileContext) {
+        let anonymous_class_name = ctx.name_identifier.take();
+
+        // 1. Let env be the LexicalEnvironment of the running execution context.
+        // 2. Let classEnv be NewDeclarativeEnvironment(env).
+        ctx.exe
```

Comment:
It'd be good to add a comment that, although the spec text doesn't set `classEnv` as the current lexical environment here, no JS code can run before it's set.

Reply:
Added a comment.